rfcs/0000-observability-api.md

```diff
@@ -0,0 +1,223 @@
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a new Kuadrant observability API. Providing a way for users to enable or disable different pieces of the observability stack.
+# Motivation
+[motivation]: #motivation
+
+Users of Kuadrant (Platform engineers and or Site reliability engineers) will want a way to be able to opt in or opt out of different aspects of observability. Providing users this ability allows them to integrate Kuadrant observability pieces with their own observability stack or Openshift user workload monitoring. Resulting in a fully comprehensive observability view all configured in one location.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The observability api, will allow a user to choose different parts of the offered Kuadrant observability. This provides more freedom to the user rather then the all or nothing current approach for certain aspects or the manual entry in multiple locations approach. With the new way the user will only have to fill in their configuration in one location.
+
+### Potential configurations
+
+The different aspects a user might want to modify could be the following:
+
+| Observability piece   | Kuadrant component                                        | Options                                                                                                                       |
+|-----------------------|-----------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------|
+| Logging               | Kuadrant operator, Authorino operator, Limitador Operator | **component**  **logLevel**:  **logMode**:                                                           |
+| Tracing               | Authorino operator, Limitador operator                    | **component** , **endpoint** , **tags**  , **insecure**, **strageyRules**  |
+| Metrics               | Kuadrant operator, Authorino, Limitador, DNS Operator     | **component** , **enableService** , **port** , **deep**                                            |
+| Alerts *              | Kuadrant operator, Authorino, Limitador                   | **namespace** , **component**, **enable**                                                                 |
+| Dashboards *          | Kuadrant operator, Authorino, Limitador                   | **namespace** , **component**, **enable**                                                                 |
+| Other 3rd Party *     | e.g Kiali                                                 | **enable** bool                                                                                                               |
+
+###### **Note**: Observability pieces with a * denotes these are post v1 milestone
+
+### Example CR with everything
```

Comment:
I wonder if it would clarify expectations better if you say this example is an ideal example if every component had every option available, and was configurable via the same type of API?
Reiterating about the evolution of the spec via versioning could re-inforce that.
Perhaps a set of guidelines the spec could follow is enough at this stage (which is kind of what is shown anyways)

Reply:
Sure i can do that